runtime/bcutil/ClassFileOracle.cpp

```diff
@@ -521,6 +529,60 @@ ClassFileOracle::walkAttributes()
 	}
 }
 
+void
+ClassFileOracle::walkRecordComponents(J9CfrAttributeRecord *attrib)
+{
+	ROMClassVerbosePhase v(_context, ClassFileAttributesRecordAnalysis);
+
+	if (0 == attrib->numberOfRecordComponents) {
+		return;
+	}
+
+	_recordComponentCount = attrib->numberOfRecordComponents;
+
+	_recordComponentsInfo = (RecordComponentInfo *) _bufferManager->alloc(_recordComponentCount * sizeof(RecordComponentInfo));
+	if (NULL == _recordComponentsInfo) {
+		Trc_BCU_ClassFileOracle_OutOfMemory((U_32)getUTF8Length(getClassNameIndex()), getUTF8Data(getClassNameIndex()));
+		_buildResult = OutOfMemory;
+		return;
+	}
+	memset(_recordComponentsInfo, 0, _recordComponentCount * sizeof(RecordComponentInfo));
+
+	for (U_16 i = 0; i < _recordComponentCount; i++) {
+		J9CfrRecordComponent* recordComponent = &attrib->recordComponents[i];
+
+		markConstantUTF8AsReferenced(recordComponent->nameIndex);
+		_recordComponentsInfo[i].nameIndex = recordComponent->nameIndex;
+		markConstantUTF8AsReferenced(recordComponent->descriptorIndex);
+		_recordComponentsInfo[i].descriptorIndex = recordComponent->descriptorIndex;
+
+		/* track record component attributes */
+		for (U_16 j = 0; j < recordComponent->attributesCount; j++) {
+			J9CfrAttribute* recordComponentAttr = recordComponent->attributes[j];
+			switch(recordComponentAttr->tag) {
+			case CFR_ATTRIBUTE_Signature: {
+				J9CfrAttributeSignature *signature = (J9CfrAttributeSignature *) recordComponentAttr;
+				markConstantUTF8AsReferenced(signature->signatureIndex);
+				_recordComponentsInfo[i].hasGenericSignature = true;
+				_recordComponentsInfo[i].genericSignatureIndex = signature->signatureIndex;
+				break;
+			}
+			case CFR_ATTRIBUTE_RuntimeVisibleAnnotations: {
+				_recordComponentsInfo[i].annotationsAttribute = (J9CfrAttributeRuntimeVisibleAnnotations *) recordComponentAttr;
+				break;
+			}
+			case CFR_ATTRIBUTE_RuntimeVisibleTypeAnnotations: {
+				_recordComponentsInfo[i].typeAnnotationsAttribute = (J9CfrAttributeRuntimeVisibleTypeAnnotations *) recordComponentAttr;
+				break;
+			}
+			default:
+				Trc_BCU_ClassFileOracle_walkFields_UnknownAttribute((U_32)attrib->tag, (U_32)getUTF8Length(attrib->nameIndex), getUTF8Data(attrib->nameIndex), attrib->length);
+				break;
```

Comment:
Should this set `_buildResult` to indicate a failure?

Reply:
The practice for walkMethods, walkFields and walkAttributes is to skip the unknown attribute instead of throwing an error, I am following along with those.

Reply:
Good point.  Unknown attributes are fine and are just passed along as data